File: typescipt/existing/src/language/expressions/functions/secondFunction.ts



namespace Lexy.Compiler.Language.Expressions.Functions;

public class SecondFunction : SingleArgumentFunction
{
   public const string Name = "SECOND";

   protected override string FunctionHelp => $"'{Name} expects 1 argument (Date)";

   protected override VariableType ArgumentType => PrimitiveType.Date;
   protected override VariableType ResultType => PrimitiveType.Number;

   private SecondFunction(Expression valueExpression, SourceReference reference)
     : base(valueExpression, reference)
   {
   }

   public static ExpressionFunction Create(SourceReference reference, Expression expression)
   {
     return new SecondFunction(expression, reference);
   }
}

File: src/Lexy.Compiler/Language/Expressions/Functions/SecondFunction.cs
using Lexy.Compiler.Language.VariableTypes;

namespace Lexy.Compiler.Language.Expressions.Functions;

public class SecondFunction : SingleArgumentFunction
{
    public const string Name = "SECOND";

    protected override string FunctionHelp => $"'{Name} expects 1 argument (Date)";

    private SecondFunction(Expression valueExpression, ExpressionSource source)
        : base(Name, valueExpression, source, PrimitiveType.Date, PrimitiveType.Number)
    {
    }

    public static FunctionCallExpression Create(ExpressionSource source, Expression expression)
    {
        return new SecondFunction(expression, source);
    }
}